server/eclipse-project/src/test/java/au/org/democracydevelopers/corla/endpoint/GetAssertionsTests.java

```diff
@@ -68,186 +72,220 @@
+
+  /**
+   * Endpoint for getting assertions.
+   */
+  final String getAssertionsEndpoint = "/raire/get-assertions";
+
+  /**
+   * JSON suffix.
+   */
+  final String json = "json";
+
+  /**
+   * CSV suffix.
+   */
+  final String csv = "csv";
+
+  /**
+   * Wiremock server for mocking the raire service.
+   * (Note the default of 8080 clashes with the raire-service default, so this is different.)
+   */
+  final WireMockServer wireMockRaireServer = new WireMockServer(8111);
+
+  /**
+   * Base url - this is set up to use the wiremock server, but could be set here to wherever you have the
+   * raire-service running to test with that directly.
+   */
+  private static String baseUrl;
+
+  /**
+   * The Properties that will be mocked in Main, specifically for the RAIRE_URL.
+   */
+  private static final Properties mockProperties = new Properties();
+
+  /**
+   * Database init.
+   */
+  @BeforeClass
+  public static void beforeAll() {
+    postgres.start();
+    Persistence.setProperties(createHibernateProperties(postgres));
+
+    var containerDelegate = new JdbcDatabaseDelegate(postgres, "");
+    // Load in the counties data, actually just for basic setup such as DoSDashboard.
+    ScriptUtils.runInitScript(containerDelegate, "SQL/co-counties.sql");
+  }
+
+  /**
+   * Initialise mocked objects prior to the first test.
+   */
+  @BeforeClass
+  public void initMocks() {
+    MockitoAnnotations.openMocks(this);
+
+    boulderIRVContestResult.setAuditReason(AuditReason.COUNTY_WIDE_CONTEST);
+    boulderIRVContestResult.setBallotCount(100000L);
+    boulderIRVContestResult.setWinners(Set.of("Aaron Brockett"));
+    boulderIRVContestResult.addContests(Set.of(boulderMayoralContest));
+
+    tinyIRVContestResult.setAuditReason(AuditReason.COUNTY_WIDE_CONTEST);
+    tinyIRVContestResult.setBallotCount(10L);
+    tinyIRVContestResult.setWinners(Set.of("Alice"));
+    tinyIRVContestResult.addContests(Set.of(tinyIRVExample));
+
+    wireMockRaireServer.start();
+    baseUrl = wireMockRaireServer.baseUrl();
+    configureFor("localhost", wireMockRaireServer.port());
+    // Mock the above-initialized URL for the RAIRE_URL property in Main.
+    mockProperties.setProperty(RAIRE_URL, baseUrl);
+
+    stubFor(post(urlEqualTo(getAssertionsEndpoint + "-" + csv))
+        .willReturn(aResponse()
+            .withStatus(HttpStatus.SC_OK)
+            .withHeader("Content-Type", "application/octet-stream")
+            .withBody("Test csv")));
+    stubFor(post(urlEqualTo(getAssertionsEndpoint + "-" + json))
+        .willReturn(aResponse()
+            .withStatus(HttpStatus.SC_OK)
+            .withHeader("Content-Type", "application/json")
+            .withBody("Test json")));
+
+  }
+
+  @AfterClass
+  public void closeMocks() {
+    wireMockRaireServer.stop();
+  }
+
+
+  /**
+   * Test data for the two valid IRV contests, with json and csv.  This gives the successfully-sanitized version of
+   * the contest names.
```

Comment:
JSON/CSV extensions are defined as constants but the actual values are used here.